### ISSUE TITLE
Not working on M1 chips

### ISSUE THREAD

**Issue by mpetrunic** [anandsuresh/sse4_crc32#84]
dyld: Symbol not found: __Z18sse42_is_availablev
  Referenced from: /Users/****/Projects/lodestar/node_modules/sse4_crc32/build/Release/crc32c.node
  Expected in: flat namespace

version: 6.0.1

**Comment by anandsuresh** [comment]
Don't have access to an M1 chip at the moment. Can you please paste the output of `npm run debug`?

**Comment by jtbandes** [comment]
I'm running into this issue as well.

In case it helps, here's the output of `nm node_modules/sse4_crc32/build/Release/crc32c.node`. Notice that the requested symbol is marked as `U` for undefined: `U __Z18sse42_is_availablev`. This makes me think the function is being stripped out by the linker, or something like that... Maybe the function need some attribute added like `NODE_MODULE_EXPORT`?

```
00000000000037e4 s GCC_except_table0
00000000000038c8 s GCC_except_table1
0000000000003b20 s GCC_except_table10
0000000000003b30 s GCC_except_table12
0000000000003b40 s GCC_except_table13
0000000000003b5c s GCC_except_table15
0000000000003bc4 s GCC_except_table16
0000000000003bd4 s GCC_except_table17
0000000000003910 s GCC_except_table2
0000000000003c2c s GCC_except_table20
0000000000003c64 s GCC_except_table21
0000000000003920 s GCC_except_table3
0000000000003950 s GCC_except_table4
0000000000003a34 s GCC_except_table5
0000000000003aac s GCC_except_table6
0000000000003ac0 s GCC_except_table7
0000000000003aec s GCC_except_table9
                 U __Unwind_Resume
00000000000030b8 T __Z10table_initv
00000000000026b4 T __Z11__napi_initP10napi_env__P12napi_value__
                 U __Z15sse42_calculatejPKcm
0000000000003220 T __Z15table_calculatejPKcm
                 U __Z18sse42_is_availablev
00000000000030b0 T __Z18table_is_availablev
000000000000223c T __Z4initN4Napi3EnvENS_6ObjectE
0000000000001364 T __Z9sse42_crcRKN4Napi12CallbackInfoE
0000000000001c7c T __Z9table_crcRKN4Napi12CallbackInfoE
000000000000271c t __ZL16_register_crc32cv
00000000000081a0 d __ZL7_module
0000000000002f90 t __ZN4Napi12CallbackInfoC2EP10napi_env__P20napi_callback_info__
0000000000002728 t __ZN4Napi5Error3NewEP10napi_env__
00000000000029cc t __ZN4Napi5Error5FatalEPKcS2_
00000000000029e4 t __ZN4Napi5ErrorD0Ev
0000000000002944 t __ZN4Napi5ErrorD1Ev
0000000000002ea4 T __ZN4Napi7details12CallbackDataIPFNS_5ValueERKNS_12CallbackInfoEES2_E7WrapperEP10napi_env__P20napi_callback_info__
0000000000002edc t __ZN4Napi7details12WrapCallbackIZNS0_12CallbackDataIPFNS_5ValueERKNS_12CallbackInfoEES3_E7WrapperEP10napi_env__P20napi_callback_info__EUlvE_EEP12napi_value__T_
0000000000002580 t __ZN4Napi8Function3NewIPFNS_5ValueERKNS_12CallbackInfoEEEES0_P10napi_env__T_PKcPv
0000000000001924 t __ZN4Napi9TypeError3NewEP10napi_env__PKc
0000000000002ce0 t __ZN4Napi9TypeErrorD0Ev
0000000000001aa8 t __ZN4Napi9TypeErrorD1Ev
0000000000002b10 t __ZNK4Napi15ObjectReference3GetEPKc
0000000000002d58 t __ZNK4Napi5Error26ThrowAsJavaScriptExceptionEv
0000000000002a5c t __ZNK4Napi5Error4whatEv
0000000000001b30 t __ZNK4Napi6String9Utf8ValueEv
                 U __ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE6resizeEmc
                 U __ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEE7reserveEm
                 U __ZNSt3__112basic_stringIcNS_11char_traitsIcEENS_9allocatorIcEEEC1ERKS5_
                 U __ZNSt9exceptionD2Ev
                 U __ZSt9terminatev
0000000000004058 s __ZTIN4Napi15ObjectReferenceE
0000000000004080 s __ZTIN4Napi5ErrorE
0000000000004048 s __ZTIN4Napi9ReferenceINS_6ObjectEEE
00000000000040b8 s __ZTIN4Napi9TypeErrorE
0000000000004070 s __ZTISt9exception
0000000000003e10 s __ZTSN4Napi15ObjectReferenceE
0000000000003e02 s __ZTSN4Napi5ErrorE
0000000000003e29 s __ZTSN4Napi9ReferenceINS_6ObjectEEE
0000000000003df0 s __ZTSN4Napi9TypeErrorE
0000000000003e48 s __ZTSSt9exception
                 U __ZTVN10__cxxabiv117__class_type_infoE
                 U __ZTVN10__cxxabiv120__si_class_type_infoE
                 U __ZTVN10__cxxabiv121__vmi_class_type_infoE
00000000000040d0 s __ZTVN4Napi5ErrorE
00000000000040f8 s __ZTVN4Napi9TypeErrorE
0000000000002ecc t __ZZN4Napi7detailsL10AttachDataINS0_12CallbackDataIPFNS_5ValueERKNS_12CallbackInfoEES3_EEEE11napi_statusP10napi_env__P12napi_value__PT_ENUlSC_PvSH_E_8__invokeESC_SH_SH_
                 U __ZdaPv
                 U __ZdlPv
                 U __Znam
                 U __Znwm
0000000000002b04 t ___clang_call_terminate
                 U ___cxa_allocate_exception
                 U ___cxa_begin_catch
                 U ___cxa_end_catch
                 U ___cxa_free_exception
                 U ___cxa_throw
                 U ___gxx_personality_v0
0000000000008198 d __dyld_private
                 U _napi_close_escapable_handle_scope
                 U _napi_close_handle_scope
                 U _napi_create_double
                 U _napi_create_error
                 U _napi_create_external
                 U _napi_create_function
                 U _napi_create_reference
                 U _napi_create_string_utf8
                 U _napi_create_symbol
                 U _napi_create_type_error
                 U _napi_define_properties
                 U _napi_delete_reference
                 U _napi_escape_handle
                 U _napi_fatal_error
                 U _napi_get_and_clear_last_exception
                 U _napi_get_boolean
                 U _napi_get_buffer_info
                 U _napi_get_cb_info
                 U _napi_get_last_error_info
                 U _napi_get_named_property
                 U _napi_get_reference_value
                 U _napi_get_typedarray_info
                 U _napi_get_undefined
                 U _napi_get_value_string_utf8
                 U _napi_get_value_uint32
                 U _napi_is_buffer
                 U _napi_is_exception_pending
                 U _napi_module_register
                 U _napi_open_escapable_handle_scope
                 U _napi_open_handle_scope
                 U _napi_set_named_property
                 U _napi_throw
                 U _napi_typeof
                 U _strlen
00000000000081e8 S _table
                 U dyld_stub_binder
```

**Comment by StBlackwood** [comment]
I am also facing this issue, anyone found a solution for this?

**Comment by anandsuresh** [comment]
Sorry guys. I don't have access to an M1 Mac to be able to test this out. I'm happy to help get any PRs merged.

**Comment by jtbandes** [comment]
BTW, I created a  faster pure-javascript implementation at https://github.com/foxglove/crc

**Comment by mikaelsahlstrom** [comment]
See #89 for a solution for this.

**Comment by anandsuresh** [comment]
Merged and published v7.0.0 with support for Apple M1 chips. Thanks @mikaelsahlstrom for the pull request.

**Comment by Brooooooklyn** [comment]
@mpetrunic https://github.com/napi-rs/node-rs/tree/main/packages/crc32 supports M1 chips with `neon`  instructions
